backend/cmm_helpers.ml

```diff
@@ -1568,8 +1568,59 @@ let call_cached_method obj tag cache pos args args_type result (apos, mode) dbg
 
 (* Allocation *)
 
-let make_alloc_generic ?(scannable_prefix = Scan_all) ~mode set_fn dbg tag
-    wordsize args =
+let mixed_block_size memory_chunks =
+  (* CR layouts 5.1: When we pack int32s/float32s more efficiently, this code
+     will need to change. *)
+  List.fold_left
+    (fun acc memory_chunk ->
+      match memory_chunk with
+      | Byte_unsigned | Byte_signed | Sixteen_unsigned | Sixteen_signed ->
+        Misc.fatal_errorf
+          "Fields with memory chunk %s are not allowed in mixed blocks"
+          (Printcmm.chunk memory_chunk)
+      | Thirtytwo_unsigned | Thirtytwo_signed ->
+        (* Int32s are currently stored using a whole word *)
+        acc + 1
+      | Single _ | Double ->
+        (* Float32s are currently stored using a whole word *)
+        if size_float <> size_addr
+        then
+          Misc.fatal_error
+            "Unable to compile mixed blocks on a platform where a float is not \
+             the same width as a value.";
+        acc + 1
+      | Word_int | Word_val -> acc + 1
+      | Onetwentyeight_unaligned | Onetwentyeight_aligned -> acc + 2)
+    0 memory_chunks
+
+let alloc_generic_set_fn block ofs newval memory_chunk dbg =
+  let generic_case () =
+    let addr = array_indexing log2_size_addr block ofs dbg in
+    Cop (Cstore (memory_chunk, Assignment), [addr; newval], dbg)
```

Comment:
Not sure if it matters, but this can use `Initialization`

Reply:
Done